Initial support for Oculus GO / Gear VR controller

otri opened the pull request:
Detects which controller side, and detects when the trigger is held.  Works without Avatar.
(There is a known bug with Avatar system on Oculus GO holding up its implementation and testing.)

This is a work in progress.  More work needed on the rest of the mappings, but good enough for enabling Arc Teleporter to work when using "Trigger = TELEPORT"

I hope to enable the rest of the mappings later when I get time, but that might be awhile.

raeldor commented:
I also made a similar change to try and get this to work, but also noticed it's teleporting to the wrong place.  This could be because I'm attaching to the LocalAvatar prefab that's then attached in the Social Starter sample to the OVRPlayerController Tracking Space.  You haven't seen issues with it teleporting to the wrong place with the Oculus Go have you?  Thanks.

otri commented:
Avatar System currently has a couple bugs with it that make it really annoying to use in practice, so I had to give up trying to use it for now.

For my set-up it looks like this:
* VRRoot/OVRCameraRig/TrackingSpace/LeftHandAnchor/TrackedRemote/OculusGoControllerModel
* VRRoot/OVRCameraRig/TrackingSpace/RightHandAnchor/TrackedRemote/OculusGoControllerModel

1. Use a **VRRoot** empty game object to hold everything.
1. Added the **OVRCameraRig** prefab to VRRoot.
1. Added the **TrackedRemote** prefab to the Left/Right Hand Anchor
1. Attached **Arc Teleporter** to the OculusGoControllerModel.
1. Make Arc Teleporter's **VRPlayArea** point at the OVRCameraRig.  This gives the correct teleport behavior.

When rigging up the LocalAvatar, I used the OVRCameraRig as its parent, and dropped its position to compensate for the [1.5m floating bug found here](https://forums.oculusvr.com/developer/discussion/58964/android-avatars-floating-at-1-5m).

raeldor commented:
Appreciate the fast response.  Will take a look at this, thank you.  The social starter seems to have OvrPlayerController as the effective VRRoot, so I set the vrPlayArea property to that and it did help with the initial positioning.  I attached my ArcTeleporter script to the LocalAvatar prefab's hand_right; maybe that was a bad move, but it seems to work.

I just found my bad positioning is caused by the Social Starter sample having a main camera.  If I put the following code into the GetLoggedInUserCallback() it seems to work now...

        // set proper base and camera
        ArcTeleporter teleporter = localAvatar.transform.GetComponentInChildren<ArcTeleporter>();
        teleporter.vrPlayArea = this.transform;
        teleporter.vrCamera = this.transform.Find("OVRCameraRig/TrackingSpace/CenterEyeAnchor");

Looks like there's a number of compatibility issues with the Avatar SDK and Oculus Go.  Not sure how these guys built Oculus Rooms; maybe not in Unity.  Anyways, they say a lot will be fixed in the next Avatar SDK release...

https://forums.oculusvr.com/developer/discussion/67782/oculususerid-confusion-social-sample-cant-get-personalized-avatars-to-work#latest
